inc/ILogOutputter.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "LogData.hpp"
+
+namespace weblog {
+
+/**
+ * @brief Interface for log outputters.
+ *
+ * The ILogOutputter interface is used to implement different log outputters.
+ * The log outputters must implement the log function to log messages and the
+ * getFormattedMessage to format the message.
+ * The copy ctor and copy assignment operator are implemented to silence clang-tidy warnings.
+ */
+class ILogOutputter {
+
+public:
+	ILogOutputter() {};
+	virtual ~ILogOutputter() {};
+
+	virtual void log(const LogData& logData) = 0;
+
+private:
+	ILogOutputter(const ILogOutputter& ref) { (void)ref; };
```

Comment:
This should be an interface with no actual implementation.

It originates from the clang-tidy warnings. In newer code you would write
```
virtual ~ILogOutputter() = default
ILogOutputter(const ILogOutputter& ref) = delete;
ILogOutputter& operator=(const ILogOutputter& rhs) = delete;
```
Since we can't use the default or delete keywords (C++11) I need to make them private for them to be unusable. Since I define them I also need to implement, which I actually don't want. So i just have dummy implementations.
I decided against moving them to a .cpp file since they don't actually implement something

Reply:
Since it is now a base class, moved the implementation there